src/ray/raylet/scheduling_resources.cc

```diff
@@ -104,6 +105,21 @@ bool ResourceSet::AddResources(const ResourceSet &other) {
   return true;
 }
 
+// Perform an outer join.
+void ResourceSet::OuterJoin(const ResourceSet &other) {
+  for (const auto &resource_pair : other.GetResourceMap()) {
+    const std::string &resource_label = resource_pair.first;
+    const double &resource_capacity = resource_pair.second;
+    if (resource_capacity_.count(resource_label) == 0) {
+      // Add the new label if not found.
+      RAY_CHECK(AddResource(resource_label, resource_capacity));
```

Comment:
I realize that `AddResource` wasn't added by this PR, but just looking at it now, I find the implementation a little fishy (it does an update in the map that always succeeds and then returns true). I think I would either add a check in `AddResource` by using `unordered_map::insert`, or I would remove it entirely and just always call `resource_capacity_[resource_label] += resource_capacity` here.